### PR TITLE
6782: usage stats improvements

### PR DESCRIPTION
**Pull request by bozana**
This PR considers the following issues:
https://github.com/pkp/pkp-lib/issues/6782
https://github.com/pkp/pkp-lib/issues/6895
https://github.com/pkp/pkp-lib/issues/6781
https://github.com/pkp/pkp-lib/issues/7318
https://github.com/pkp/pkp-lib/issues/5543

The PRs and commits explanations is provided in the PKP-LIB PR: https://github.com/pkp/pkp-lib/pull/8050

### CODE REVIEW

**Review by asmecher — COMMENTED** [review]
@bozana, just a few initial comments while I figure out how this all fits together!

File: plugins/themes/default/DefaultThemePlugin.inc.php [hunk 1]
```diff
@@ -234,4 +254,12 @@ public function getDescription()
     {
         return __('plugins.themes.default.description');
     }
+
+    /**
+     * Get the context for inclusion of usage stats display related JavaScripts in the submission view page
+     */
+    protected function getSubmissionViewContext(): string
+    {
+        return 'frontend-article-view';
```
[the remark this answers is not in the document]
Comment (bozana) [reply plugins/themes/default/DefaultThemePlugin.inc.php:263]: Yes. This is different for OJS, OMP and OPS, but maybe I can check the application in the ThemePlugin and use one accordingly...

[the remark this answers is not in the document]
Comment (bozana) [reply plugins/themes/default/DefaultThemePlugin.inc.php:263]: :+1 I moved it to ThemePlugin. The changes that I make now I will push to my other branch (6782-3)...

File: api/v1/stats/index.php [hunk 2]
```diff
@@ -26,4 +28,13 @@
...
+} elseif (strpos($requestPath, '/stats/issues')) {
+    import('api.v1.stats.issues.StatsIssueHandler');
+    return new StatsIssueHandler();
+} elseif (strpos($requestPath, '/stats/contexts')) {
+    import('lib.pkp.api.v1.stats.contexts.PKPStatsContextHandler');
+    return new PKPStatsContextHandler();
+} elseif (strpos($requestPath, '/stats/sushi')) {
+    import('api.v1.stats.sushi.StatsSushiHandler');
+    return new StatsSushiHandler();
 }
```
Comment (asmecher) [inline api/v1/stats/index.php:40]: This (and probably other handlers here) needs to handle errors in the default case. Requesting e.g. `http://localhost/git/ojs/index.php/index/api/v1/stats/` causes an error:
```
[20-Jul-2022 01:20:22 UTC] PHP Fatal error:  Uncaught Error: Call to a member function getApp() on int in /home/asmecher/git/ojs-main/lib/pkp/classes/core/APIRouter.inc.php:128
Stack trace:
#0 /home/asmecher/git/ojs-main/lib/pkp/classes/core/Dispatcher.inc.php(163): PKP\core\APIRouter->route()
#1 /home/asmecher/git/ojs-main/lib/pkp/classes/core/PKPApplication.inc.php(379): PKP\core\Dispatcher->dispatch()
#2 /home/asmecher/git/ojs-main/index.php(70): PKP\core\PKPApplication->execute()
#3 {main}
  thrown in /home/asmecher/git/ojs-main/lib/pkp/classes/core/APIRouter.inc.php on line 128
```

**Comment by bozana** [comment]
closing the pr -- it was for the second review round...

**Pull request closed without merging**